### PR TITLE
feat(Dropdown): Controlled Dropdown / Menu Integration Branch

### PR DESCRIPTION
**Pull request by saurabhdaware**
- [x] #1139 
- [x] #1152 
- [x] ~#1161~ (Merging this separately)
- [x] Changeset and Migration Guide

### CODE REVIEW

File: .changeset/clever-doors-compare.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+'@razorpay/blade': major
+---
+
+feat(Dropdown): Controlled Dropdown and Button Trigger
```
Comment (divyanshu013) [inline .changeset/clever-doors-compare.md:5]: A brief explanation of the major release, eg:

```suggestion
feat(Dropdown): Controlled Dropdown and Button Trigger
- Adds APIs to seamlessly build controlled dropdown
- Adds ways to use dropdown as a menu - using a dropdown button as a trigger
- Removes `isDefaultSelected` from `ActionListItem` *(see migration guide)*
```

**Review by divyanshu013 — DISMISSED** [review]
:shipit:

File: .changeset/clever-doors-compare.md (hunk 1 above)
Comment (saurabhdaware) [reply .changeset/clever-doors-compare.md:5]: done

**Review by divyanshu013 — APPROVED** [review]
:shipit:

**Pull request merged**